dnscrypt-proxy/local-doh.go

```diff
@@ -98,7 +98,11 @@ func (handler localDoHHandler) ServeHTTP(writer http.ResponseWriter, request *ht
 }
 
 func (proxy *Proxy) localDoHListener(acceptPc *net.TCPListener) {
-	defer acceptPc.Close()
+	defer func(acceptPc *net.TCPListener) {
+		if acceptPc != nil {
```

Comment:
In this case, if for some reason `acceptPc` is `nil` then there's no point doing anything in `localDoHListener`. The simpler solution here is an early `return`.